### DISCUSSION TITLE
Error creating local backup with mysqldump

### DISCUSSION THREAD

**Discussion by Merlin04** [planetscale/discussion#194] [Q&A]
I'm trying to create a local backup of my PlanetScale database using the "Export with mysqldump" option in JetBrains DataGrip:

![pscale-dump](https://user-images.githubusercontent.com/11800751/168648301-0d4c58b0-1050-4279-85c0-60710938cffa.png)

However, my database has a table with very large data, which appears to be causing it to fail:

```
[2022-05-16 10:18:59] /usr/bin/mysqldump [database name] --result-file=/home/[...]/[...]-2022_05_16_10_18_58-dump.sql --user=[user] --host=[host] --port=3306
Warning: A partial dump from a server that has GTIDs will by default include the GTIDs of all transactions, even those that changed suppressed parts of the database. If you don't want to restore GTIDs, pass --set-gtid-purged=OFF. To make a complete dump, pass --all-databases --triggers --routines --events. 
mysqldump: Couldn't execute 'SELECT /*!40001 SQL_NO_CACHE */ * FROM `[big table name]`': vttablet: rpc error: code = ResourceExhausted desc = grpc: trying to send message larger than max (94857006 vs. 67108864) (1203)
[2022-05-16 10:19:03] mysqldump process finished
```

How can I successfully dump the database to a local file?

**Comment by fooforge** [comment]
Hey @Merlin04,

I haven't had the chance to try out dumping data with JetBrain's DataGrip yet, but it's on my list. In the mean time, is dumping through our [command-line utility `pscale`](https://planetscale.com/features/cli) an option?

That only requires a `pscale database dump [database] [branch]` and will result in a new folder being created in your current working directory with (at least) two SQL dumps per table: One for the table's schema and one for the actual data stored inside the table.

**Reply by davidmcl** [reply]
While @fooforge is correct, that "dump" isn't a standard mysql workbench dump you would get from an export, so that it courld be used to import via workbench.

It might be OK to restore using `pscale database restore-dump` it's pretty worthless (very time consuming) to try and use that dump to import into another mysql instance on RDS, GCloud, or a local machine for testing.

I would like to see them support an export via workbench and/or have an option in pscale like `dump-for-import` to allow a workbench import.

As it is, it gives one pause to consider a production implementation without the ability to use the standard tools and get data exported how  we want it.

David

**Reply by Merlin04** [reply]
I was able to use the pscale dump command for my purposes (didn't know it existed) but I completely agree that it would be great to be able to dump the DB in a vendor-agnostic way.
